### PR TITLE
feat(ingest): Support alternative authentication in sql ingestion

### PR DESCRIPTION
**Pull request by pedro93**
Makes username/password/database config properties optional to allow sqlachemy to connect with alternative authentication methods such as kerberos.

This was tested by specifying only the mandatory host_port config and then passing extra required properties through the `options` map which is specific to each `<SQL database, Auth Method>` combination.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Review by shirshanka — APPROVED** [review]
LGTM. Thanks @pedro93!
Will wait for build to turn green before merging it in.

**Comment by pedro93** [comment]
Build failed but does not seem related:
```
fetched 4613 kB in 3s (1723 kB/s)
Reading package lists...
+ sudo apt-get install -y librdkafka-dev python-ldap libldap2-dev libsasl2-dev ldap-utils
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package python-ldap
Error: Process completed with exit code 100.
```

**Comment by hsheth2** [comment]
Sorry about that - I'm hoping https://github.com/linkedin/datahub/pull/2151/commits/9bbee3f3c645b54c6b68a44bd5c7f721d9859632 will fix it. I'm just waiting for CI on #2151, after which you'll probably need to rebase.

**Pull request merged**
